App/Sources/Application/DI/Auth/AppComponent+Auth.swift

```diff
@@ -0,0 +1,61 @@
+import NeedleFoundation
+import Service
+
+public extension AppComponent {
+    var localAuthDataSource: any LocalAuthDataSource {
+        shared {
+            LocalAuthDataSourceImpl(keychain: keychain)
+        }
+    }
+
+    var remoteAuthDataSource: any RemoteAuthDataSource {
+        shared {
+            RemoteAuthDataSourceImpl(keychian: keychain)
+        }
+    }
+
+    var authRepository: any AuthRepository {
+        shared {
+            AuthRepositoryImpl(
+                remoteAuthDataSource: remoteAuthDataSource,
+                localAuthDataSource: localAuthDataSource
+            )
+        }
+    }
+
+    var loginUseCase: any LoginUseCase {
+        shared {
+            LoginUseCaseImpl(authRepository: authRepository)
+        }
+    }
+
+    var reissueTokenUseCase: any ReissueTokenUseCase {
+        shared {
+            ReissueTokenUseCaseImpl(authRepository: authRepository)
+        }
+    }
+
+    var logoutUseCase: any LogoutUseCase {
+        shared {
+            LogoutUseCaseImpl(authRepository: authRepository)
+        }
+    }
+
+    var withdrawUseCase: any WithdrawalUseCase {
+        shared {
+            WithdrawalUseCaseImpl(authRepository: authRepository)
+        }
+    }
```

Comment:
```suggestion
    var withdrawalUseCase: any WithdrawalUseCase {
        shared {
            WithdrawalUseCaseImpl(authRepository: authRepository)
        }
    }
```
여기도 withdrawal로 해야하는거 아닌가요?